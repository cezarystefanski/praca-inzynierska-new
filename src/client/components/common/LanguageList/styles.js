import { withStyles } from 'styles/themes/ExampleTheme/ExampleTheme';

export default withStyles(({ color }) => ({
  inactiveLanguage: {
    color: color.grey,
    cursor: 'not-allowed',
    textDecoration: 'none',
  },
  languageElement: {
    cursor: 'pointer',
    marginLeft: '0.25rem',
    textDecoration: 'underline',

    ':not(:first-child)': {
      ':before': {
        color: color.white,
        content: '\'|\'',
        marginRight: '0.25rem',
      },
    },
  },
  languageList: {
    display: 'flex',
    listStyle: 'none',
    margin: 0,
    padding: 0,
  },
  resettedButton: {
    background: 'none',
    color: 'inherit',
    border: 'none',
    padding: '0',
    font: 'inherit',
    cursor: 'inherit',
    outline: 'inherit',
    textDecoration: 'inherit',
  },
}));
