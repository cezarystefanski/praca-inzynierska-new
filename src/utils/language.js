import {
  LOCAL_STORAGE_PREFIX,
  LOCAL_STORAGE_LANGUAGE,
} from 'consts/names';
import { DEFAULT_LANGUAGE } from 'consts/language';

export const getLocale = () => {
  const keyName = `${LOCAL_STORAGE_PREFIX}${LOCAL_STORAGE_LANGUAGE}`;
  const localStorageLanguage = localStorage.getItem(keyName);

  if (!localStorageLanguage) {
    const currentLanguage = navigator.language || DEFAULT_LANGUAGE;

    localStorage.setItem(keyName, currentLanguage);

    return currentLanguage;
  }

  return localStorageLanguage;
};
