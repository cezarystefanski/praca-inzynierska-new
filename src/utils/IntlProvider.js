import React from 'react';
import PropTypes from 'prop-types';

import { IntlProvider as StockProvider, addLocaleData } from 'react-intl';
import * as en from 'react-intl/locale-data/en';
import * as pl from 'react-intl/locale-data/pl';

import { DEFAULT_LANGUAGE } from 'consts/language';

import messages from 'messages/messages.json';

/* istanbul ignore next line */
if (process.env.NODE_ENV !== 'test') {
  addLocaleData([...en, ...pl]);
}

export const IntlProvider = ({ children, currentLanguage, locale }) => (
  <StockProvider
    locale={locale}
    messages={messages[currentLanguage]}
  >
    {children}
  </StockProvider>
);

IntlProvider.propTypes = {
  children: PropTypes.node,
  currentLanguage: PropTypes.string,
  locale: PropTypes.string,
};

IntlProvider.defaultProps = {
  children: null,
  currentLanguage: DEFAULT_LANGUAGE,
  locale: DEFAULT_LANGUAGE,
};
