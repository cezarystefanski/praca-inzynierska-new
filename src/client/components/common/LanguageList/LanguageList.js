import React, { Component } from 'react';

import { css } from 'styles/themes/ExampleTheme/ExampleTheme';

import { setLocale } from 'utils/language';
import { LANG_LIST } from 'consts/language';

export class LanguageListView extends Component {
  changeLanguage = (lang, currentLang) => {
    if (lang === currentLang) {
      return false;
    }

    return setLocale(lang);
  }

  render() {
    const languageValues = Object.keys(LANG_LIST);
    const { styles, currentLanguage } = this.props;

    return (
      <ul {...css(styles.languageList)}>
        {
          languageValues.map(lang => (
            <li
              {...css(
                styles.languageElement,
                lang === currentLanguage && styles.inactiveLanguage,
              )}
              key={lang}
            >
              <button
                {...css(styles.resettedButton)}
                onClick={() => this.changeLanguage(lang, currentLanguage)}
                type="button"
              >
                {LANG_LIST[lang]}
              </button>
            </li>
          ))
        }
      </ul>
    );
  }
}
