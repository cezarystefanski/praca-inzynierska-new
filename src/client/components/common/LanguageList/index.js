import { compose } from 'recompose';

import { LanguageListView as LanguageList } from './LanguageList';

import withStyles from './styles';

const enhance = compose(
  withStyles,
);

const Component = enhance(LanguageList);

export default Component;
