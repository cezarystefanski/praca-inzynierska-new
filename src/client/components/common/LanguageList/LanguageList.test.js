import React from 'react';
import { mount, shallow } from 'enzyme';
import { IntlProvider } from 'utils/IntlProvider';

import LanguageList from './index';

describe('component common/LanguageList', () => {
  const wrapper = mount(
    <IntlProvider>
      <LanguageList />
    </IntlProvider>,
  );

  it('should render correctly', () => {
    expect(wrapper.find('button'))
      .toHaveLength(2);
  });

  it('should not differ from earlier version', () => {
    const shallowWrapper = shallow(
      <LanguageList />,
    ).shallow();

    expect(shallowWrapper)
      .toMatchSnapshot();
  });
});
