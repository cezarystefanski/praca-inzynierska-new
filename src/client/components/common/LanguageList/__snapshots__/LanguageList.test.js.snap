// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component common/LanguageList should not differ from earlier version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LanguageListView
    css={[Function]}
    styles={
      Object {
        "inactiveLanguage": Object {
          "_definition": Object {
            "color": "#aaa",
            "cursor": "not-allowed",
            "textDecoration": "none",
          },
          "_len": 63,
          "_name": "inactiveLanguage_17eoe3d",
        },
        "languageElement": Object {
          "_definition": Object {
            ":not(:first-child)": Object {
              ":before": Object {
                "color": "#fff",
                "content": "'|'",
                "marginRight": "0.25rem",
              },
            },
            "cursor": "pointer",
            "marginLeft": "0.25rem",
            "textDecoration": "underline",
          },
          "_len": 162,
          "_name": "languageElement_131nfzv",
        },
        "languageList": Object {
          "_definition": Object {
            "display": "flex",
            "listStyle": "none",
            "margin": 0,
            "padding": 0,
          },
          "_len": 60,
          "_name": "languageList_mn4u4y",
        },
        "resettedButton": Object {
          "_definition": Object {
            "background": "none",
            "border": "none",
            "color": "inherit",
            "cursor": "inherit",
            "font": "inherit",
            "outline": "inherit",
            "padding": "0",
            "textDecoration": "inherit",
          },
          "_len": 152,
          "_name": "resettedButton_1wm8fde",
        },
      }
    }
    theme={
      Object {
        "color": Object {
          "grey": "#aaa",
          "icyBlue": "#ccf",
          "white": "#fff",
        },
        "font": Object {
          "sizes": Object {
            "big": "20px",
            "normal": "14px",
            "small": "12px",
          },
          "weights": Object {
            "bold": 700,
            "normal": 500,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="languageElement_131nfzv"
        >
          <button
            className="resettedButton_1wm8fde"
            onClick={[Function]}
            type="button"
          >
            english
          </button>
        </li>,
        <li
          className="languageElement_131nfzv"
        >
          <button
            className="resettedButton_1wm8fde"
            onClick={[Function]}
            type="button"
          >
            polski
          </button>
        </li>,
      ],
      "className": "languageList_mn4u4y",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "en",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="resettedButton_1wm8fde"
            onClick={[Function]}
            type="button"
          >
            english
          </button>,
          "className": "languageElement_131nfzv",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "english",
            "className": "resettedButton_1wm8fde",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "english",
          "type": "button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "pl",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="resettedButton_1wm8fde"
            onClick={[Function]}
            type="button"
          >
            polski
          </button>,
          "className": "languageElement_131nfzv",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "polski",
            "className": "resettedButton_1wm8fde",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "polski",
          "type": "button",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="languageElement_131nfzv"
          >
            <button
              className="resettedButton_1wm8fde"
              onClick={[Function]}
              type="button"
            >
              english
            </button>
          </li>,
          <li
            className="languageElement_131nfzv"
          >
            <button
              className="resettedButton_1wm8fde"
              onClick={[Function]}
              type="button"
            >
              polski
            </button>
          </li>,
        ],
        "className": "languageList_mn4u4y",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "en",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="resettedButton_1wm8fde"
              onClick={[Function]}
              type="button"
            >
              english
            </button>,
            "className": "languageElement_131nfzv",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "english",
              "className": "resettedButton_1wm8fde",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "english",
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "pl",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="resettedButton_1wm8fde"
              onClick={[Function]}
              type="button"
            >
              polski
            </button>,
            "className": "languageElement_131nfzv",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "polski",
              "className": "resettedButton_1wm8fde",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "polski",
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
